src/components/imageBackgroundRe.re

```diff
@@ -0,0 +1,182 @@
+      source::imageSource? =>
+      style::StyleRe.t? =>
+      imageStyle::StyleRe.t? =>
+      testID::string? =>
+      resizeMethod::[< | `auto | `resize | `scale]? =>
+      accessibilityLabel::string? =>
+      accessible::bool? =>
+      blurRadius::float? =>
+      capInsets::TypesRN.insets? =>
+      defaultSource::defaultSource? =>
+      onPartialLoad::(unit => unit)? =>
+      onProgress::(Event.progress => unit)? =>
+      array ReasonReact.reactElement =>
+      ReasonReact.component ReasonReact.stateless ReasonReact.noRetainedProps unit;
+  };
+  
+  module CreateComponent (Impl: ViewRe.Impl) :ImageBackgroundComponent => {
+    type imageURISource;
+    external imageURISource :
+      uri::string =>
+      bundle::string? =>
+      method::string? =>
+      headers::Js.t 'a? =>
+      body::string? =>
+      /*
+       * Be careful not to refmt this away !!!
+       * https://github.com/facebook/reason/issues/821 (resolved, not released yet)
+       *
+       * This is hot it should look (or to copy it in again ^^)
+       *  cache::[ | `default | `reload | `forceCache [@bs.as "force-cache"] | `onlyIfCached  [@bs.as "only-if-cached"]] [@bs.string]? =>
+       */
+      cache::
+        [
+          | `default
+          | `reload
+          | `forceCache [@bs.as "force-cache"]
+          | `onlyIfCached [@bs.as "only-if-cached"]
+        ]
+        [@bs.string]? =>
+      scale::float? =>
+      width::float? =>
+      height::float? =>
+      unit =>
+      imageURISource =
+      "" [@@bs.obj];
+    type imageSource =
+      | URI imageURISource
+      | Required PackagerRe.required
+      | Multiple (list imageURISource);
+    type defaultURISource;
+    external defaultURISource :
+      uri::string => scale::float? => width::float? => height::float? => unit => defaultURISource =
+      "" [@@bs.obj];
+    type defaultSource =
+      | URI defaultURISource
+      | Required PackagerRe.required;
+    type rawImageSourceJS;
+    external rawImageSourceJS : 'a => rawImageSourceJS = "%identity";
+    module Event = {
+      type t;
+      type error;
+      type progress = {
+        loaded: float,
+        total: float
+      };
+      external progress : t => progress = "nativeEvent" [@@bs.get];
+    };
+    let encodeResizeMode x =>
+      switch x {
+      | `cover => "cover"
+      | `contain => "contain"
+      | `stretch => "stretch"
+      | `repeat => "repeat"
+      | `center => "center"
+      };
+    let encodeSource (x: imageSource) =>
+      switch x {
+      | URI x => rawImageSourceJS x
+      | Required x => rawImageSourceJS x
+      | Multiple x => rawImageSourceJS (Array.of_list x)
+      };
+    let encodeResizeMethod x =>
+      switch x {
+      | `auto => "auto"
+      | `resize => "resize"
+      | `scale => "scale"
+      };
+    let encodeDefaultSource (x: defaultSource) =>
+      switch x {
+      | URI x => rawImageSourceJS x
+      | Required x => rawImageSourceJS x
+      };
+    let make
```

Comment:
This module should only contain `make`.

We don't need a functor unless there exists `Animated.ImageBackground`. The other types and utility functions should be the same as in `ImageRe.re`. If they are unaccessible due to `reis` feel free to put them in a separate file. Like `ImageCommonRe.re`.